### DISCUSSION TITLE
SearchFilter incorrectly split value with istartswith filter

### DISCUSSION THREAD

**Discussion by Arti3DPlayer** [encode/django-rest-framework#9271] [Potential Issue]
When we define filter field with `istartswith` as described in [docs](https://www.django-rest-framework.org/api-guide/filtering/#searchfilter)

`search_fields=["^address", ]`

and will try to search by string that contains spaces: `123 Main St`, it will not work, because split term in this case is working incorrectly and split string as:
`Q(address_istartwith="123) | Q(address_istartwith="Main") | Q(address_istartwith="St")` ([source](https://github.com/encode/django-rest-framework/blob/fb03dd977e277048c396d9285c20db3cb55474ca/rest_framework/filters.py#L155))

 Expected behaviour to generate query like:
`Q(address_istartwith="123 Main St")` 

Version:
`djangorestframework==3.14.0`

**Comment by lovelydinosaur** [comment]
So... I don't think it's necessarily(?) that obvious which behaviour I'd expect, and might depend on the UI.

**Reply by Arti3DPlayer** [reply]
As for me this behaviour does not make sense for any real example. For example: in my project I have autocomplete widget for different entities like `User names`  `Company Names` `Addresses` , where I have `^name` and this is odd when user start typing: `123` hit space and result disappear

**Comment by lovelydinosaur** [comment]
Yeah. I'd probs override 'filter_queryset' and move on with your day.
